hub/instruction_handling/storage/constraints.tex

```diff
@@ -221,31 +230,87 @@
-			\item \If $\locNextIsCurr = 1$ \Then $\refund\new_{i} = \refund_{i}$
-			\item \If $\locNextIsCurr = 0$ \Then
-				\begin{enumerate}
-					\item \If $\locCurrIsOrig = 0$ \Then
-						\[
-							\refund\new_{i} = \refund_{i} 
-							+
-							\left[ \begin{array}{crcccl}
-								+ & (1 - \locOrigIsZero) & \cdot & (\locNextIsZero - \locCurrIsZero) & \cdot & R_\text{sclear}                                 \\
-								+ & \locNextIsOrig       & \cdot & \locOrigIsZero                    & \cdot & \big(G_\text{sset} - G_\text{warmaccess}\big)   \\
-								+ & \locNextIsOrig       & \cdot & (1 -\locOrigIsZero)               & \cdot & \big(G_\text{sreset} - G_\text{warmaccess}\big) \\
-							\end{array} \right]
-						\]
-						\item \If $\locCurrIsOrig = 1$ \Then
-							\begin{enumerate}
-								\item \If $\locNextIsZero = 0$ \Then $\refund\new_{i} = \refund_{i}$
-								\item \If $\locNextIsZero = 1$ \Then $\refund\new_{i} = \refund_{i} + R_\text{sclear}$
-							\end{enumerate}
-				\end{enumerate}
-		\end{enumerate}
-\end{description}
-
+		We therefore unconditionally impose
+		\[
+			\refund\new _{i} = \refund_{i} + (1 - \cnWillRev_{i}) \cdot \locIsSstore \cdot
+			\left[ \begin{array}{cr}
+				+ & \locCleanClear \\
+				+ & \locDirtyClear \\
+				+ & \locDirtyReset \\
+			\end{array} \right]
+		\]
+		where $\locCleanClear$, $\locDirtyClear$, $\locDirtyReset$ are shorthands which we define below.
+		
+		\saNote{}
+		In the explanations provided below we use notations from the \cite{EYP} whereby
+		$v_0$ stands for the value originally in storage,
+		$v$ for the value currently in storage,
+		$v'$ for the value we are attempting to store:
+		\begin{description}
+			\item[\underline{Defining \locCleanClear{}:}]
+				we set
+				\[
+					\locCleanClear \define
+					\left[ \begin{array}{cr}
+						\cdot & \locNextNotCurr \\
+						\cdot & \locCurrIsOrig       \\
+						\cdot & \locNextIsZero       \\
+					\end{array} \right]
+					\cdot R_\text{sclear}
+				\]
+				which corresponds to
+				\begin{itemize}
+					\item $R_\text{sclear}$ whenever $[v \neq v'] \wedge [v = v_{0}] \wedge [v' = 0]$
+					\item and $0$ otherwise
+				\end{itemize}
+			\item[\underline{Defining \locDirtyClear{}:}]
+				we set
+				\[
+					\locDirtyClear \define
+					\left[ \begin{array}{cr}
+						\cdot & \locNextNotCurr \\
+						\cdot & \locCurrNotOrig \\
+					\end{array} \right]
+					\cdot
+					\locOrigNotZero
+					\cdot 
+					\left[ \begin{array}{cr}
+						- & \locCurrIsZero \\
+						+ & \locNextIsZero \\
+					\end{array} \right]
+					\cdot R_\text{sclear}
+				\]
+				which corresponds to
+				\begin{itemize}
+					\item $- R_\text{sclear}$ whenever $[v \neq v'] \wedge [v_{0} \neq v]$ and $[v_{0} \neq 0] \wedge [v  = 0]$,
+					\item $+ R_\text{sclear}$ whenever $[v \neq v'] \wedge [v_{0} \neq v]$ and $[v_{0} \neq 0] \wedge [v' = 0]$,
+					\item and $0$ otherwise.
+				\end{itemize}
+			\item[\underline{Defining \locDirtyReset{}:}]
```

Comment:
Two cases here to consider:

- `[v' != v]` and `[v' == v0]` and `[v0 == 0]` implies `[v0 != v]` ... **[OK]**
- `[v != v0]` and `[v' == v0]` and `[v0 != 0]` implies `[v != v']` ... **[OK]**

Ok, that looks good to me.